File: sdk/typescript/src/pricing.ts
/**
 * Compute the claim cost in zatoshis for a name of the given length.
 *
 * @param tiers  The pricing tiers from `StatusResult.pricing.tiers` (already in zatoshis).
 *               Index 0 = 1-char names, index 1 = 2-char, etc.
 *               Names longer than the array clamp to the last entry.
 * @param nameLength  Character length of the name (must be >= 1).
 * @returns Cost in zatoshis, or `null` if tiers is empty.
 */
export function claimCost(tiers: number[], nameLength: number): number | null {
  if (tiers.length === 0) return null;
  const idx = Math.min(Math.max(nameLength - 1, 0), tiers.length - 1);
  return tiers[idx];
}

File: sdk/dart/lib/src/pricing.dart
/// Compute the claim cost in zatoshis for a name of the given length.
///
/// [tiers] comes from `StatusResult.pricing.tiers` (already in zatoshis).
/// Index 0 corresponds to 1-char names, index 1 to 2-char names, etc.
/// Names longer than the list clamp to the last entry.
///
/// Returns `null` if [tiers] is empty.
int? claimCost(List<int> tiers, int nameLength) {
  if (tiers.isEmpty) return null;
  final idx = (nameLength - 1).clamp(0, tiers.length - 1);
  return tiers[idx];
}
